src/components/Form/Form.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useState, useEffect } from 'react';
+import _isEmpty from 'lodash/isEmpty';
+import _set from 'lodash/set';
+
+const Form = ({
+  children,
+  initialValues = {},
+  validationSchema = {},
+  onSubmit,
+  className = '',
+}) => {
+  const [formValues, setFormValues] = useState(initialValues);
+  const [touched, setTouched] = useState({});
+  const [validationErrors, setValidationErrors] = useState({});
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [formSubmitError, setFormSubmitError] = useState(null);
+
+  const handleBlur = ({ target: { name } }) => {
+    setTouched((prevState) => ({
+      ...prevState,
+      [name]: true,
+    }));
+  };
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
```

Comment:
I mean: Should we not prevent the default at all, or just allow the user to prevent default by exposing an `onSubmit` function prop?

One potentially nice thing about `submit` events is that, for short forms, they capture when the user enters text in a field and hitting the "Enter" key, iirc.

I'd hate to trap the submit events and prevent their default behavior if they're useful.